delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/resources/ResourceCommand.scala

```diff
@@ -0,0 +1,115 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.model.resources
+
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.{CompactedJsonLd, ExpandedJsonLd}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.ResourceRef
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Identity.Subject
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.ProjectRef
+import io.circe.Json
+
+/**
+  * Enumeration of resource commands
+  */
+sealed trait ResourceCommand extends Product with Serializable {
+
+  /**
+    * @return the project where the resource belongs to
+    */
+  def project: ProjectRef
+
+  /**
+    * @return the resource identifier
+    */
+  def id: Iri
+
+  /**
+    * @return the identity associated to this command
+    */
+  def subject: Subject
+
+}
+
+object ResourceCommand {
+
+  /**
+    * Command that signals the intent to create a new resource.
+    *
+    * @param id          the resource identifier
+    * @param project     the project where the resource belongs
+    * @param schema      the schema used to constrain the resource
+    * @param source      the representation of the resource as posted by the subject
+    * @param compacted   the compacted JSON-LD representation of the resource
+    * @param expanded    the expanded JSON-LD representation of the resource
+    * @param subject     the subject which created this event
+    */
+  final case class CreateResource(
+      id: Iri,
+      project: ProjectRef,
+      schema: ResourceRef,
+      source: Json,
+      compacted: CompactedJsonLd,
+      expanded: ExpandedJsonLd,
+      subject: Subject
+  ) extends ResourceCommand
+
+  /**
+    * Command that signals the intent to update an existing resource.
+    *
+    * @param id        the resource identifier
+    * @param project   the project where the resource belongs
+    * @param schemaOpt the optional schema of the resource. A None value ignores the schema from this command
+    * @param source    the representation of the resource as posted by the subject
+    * @param compacted the compacted JSON-LD representation of the resource
+    * @param expanded  the expanded JSON-LD representation of the resource
+    * @param rev       the last known revision of the resource
+    * @param subject   the subject which created this event
+    */
+  final case class UpdateResource(
+      id: Iri,
+      project: ProjectRef,
+      schemaOpt: Option[ResourceRef],
+      source: Json,
```

Comment:
If I only want to change the schema, I still have to pass the source ?

Reply:
The schema cannot be changed at all, but it is passed for the same reason as I explained before